### ISSUE TITLE
V11 ALPHA reporting mod conflict

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#611]
Looks like patch manager found a conflict...

```
Info 475.2117273: ThreadingExtension.OnBeforeSimulationFrame: First frame detected. Detours checked. Result: 1 missing detours

Info 475.2129049: The following methods were overriden by another mod:
	<Harmony> RoadBaseAI.SimulationStep with 2 parameters (RoadBaseAI, Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null)
```

> **Serialblack** 11 Jan @ 5:15am 
> is there a list of incompatible mods with this version? Either confirmed or suspected? I had it working well, but now the game freezes at the warning pop-up saying there was/is an incompatible mod, however it doesn't write a log file, or at least I can't see one anywhere, so I'm just trying to get it sorted

> **Serialblack** 6 hours ago 
no error shows, I got it to run....I still get the pop up saying something is incompatible, but at least its running....I can still send the log , if it will help you...

[TMPE.log](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/files/4054540/TMPE.log)

User has most of the same mods I do, execpt these ones (`*` denotes they had the mod enabled):

```
  * 1844440354   Klyte's Fine Road Anarchy 2
  * 1764208250   More Vehicles
  * 1844442251   Klyte's Fine Road Tool 2.0.3
  * 1645781000   Elektrix's Road Tools
  * 774449380    Catenary Replacer
  * 1890830956   Undo It!
  * 1393797695   Touch This! Tool 3
  * 1829496988   Adjust Pathfinding
  * 1651036644   Traffic Congestion Report
  * 928128676    Improved Public Transport 2 [r4.4.0]
  * 812125426    Network Extensions 2
  * 1612012531   Net Picker
  * 814903880    Tree Fire Control
  * 1312735149   Klyte Commons 1.1.7
    1632320836   Service Vehicles Manager 2.0.1
  * 639486063    Automatic Bulldoze v2
  * 1094334744   Procedural Objects 1.6.1
  * 1729576238   Date Changer
  * 934994075    Service Vehicle Selector 2 (r4.0.2)
  * 1237383751   Extended Game Options
  * 1760945529   Terrain Level Of Detail
    952542692    Airport Road Light Remover
  * 1803209875   Trees' Respiration 1.0.1
```

**Comment by pcfantasy** [comment]
@aubergine10     
I also meet a similar issue, when I use harmony to patch RoadBaseAI.UpdateLanes, TMPE report a similar conflict.
Since TMPE and My mod CSURToolBox both use harmony to patch, I think there should be no confict. Is it a false warning produced by TMPE ThreadingExtension.OnBeforeSimulationFrame in checking detour?

            harmony = HarmonyInstance.Create("CSURToolBox");
             var roadBaseAIUpdateLanes = typeof(RoadBaseAI).GetMethod("UpdateLanes", BindingFlags.Instance | BindingFlags.Public, Type.DefaultBinder, new Type[3] { typeof(ushort), typeof(NetSegment).MakeByRefType(), typeof(bool) }, null);
             var roadBaseAIUpdateLanesPostFix = typeof(CustomNetAI).GetMethod("RoadBaseAIUpdateLanesPostFix");
             harmony.ConditionalPatch(roadBaseAIUpdateLanes,
                 null,
                 new HarmonyMethod(roadBaseAIUpdateLanesPostFix));

**Comment by aubergine10** [comment]
TM:PE applies patches then checks to see if they've been applied, and if they haven't (ie. patch failed for some reason) it shows the warning and logs some details to TMPE.log.

The user who reported the issue in OP has stated that removing the **Trees Respiration** mod (by Klyte45) resolved the issue.

**Comment by pcfantasy** [comment]
@aubergine10 

I still have a question, If the other mod and TMPE both use harmony to add a prefix or postfix.

Will TMPE report conflict?

**Comment by aubergine10** [comment]
Only if for some reason TMPE can't apply it's prefix/postfix. 

Most of the patching/checking is done here: https://github.com/CitiesSkylinesMods/TMPE/blob/master/TLM/TLM/LoadingExtension.cs

@krzychu124 is probably the best person to ask for specifics.

**Comment by pcfantasy** [comment]
Ok, I use to get a conflict information when my mod and TMPE all use harmony to patch RoadBaseAI.UpdateLanes.

But only once, later I can not produce this issue, so I want to discuss this with you all.

@krzychu124

**Comment by krzychu124** [comment]
Ummm, I don't really know why regular prefixes/postfixes sometimes show conflict dialog. Which mod is it @pcfantasy ? It is possible to set priority of prefix/postfix, but I didn't tested that yet.

**Comment by pcfantasy** [comment]
@krzychu124 

It is in early version of CSUR toolbox, however I change patch to another method and release it to steam. But recently I have to patch RoadBaseAI.UpdateLanes again but seems no conflict shows.

The conflict check is to check orginal method is not been replaced, right? So technically two prefix or postfix will not report conflict, right?

**Comment by krzychu124** [comment]
Yes theoretically there shouldn't be any issue when you add prefix or postfix. I'm not sure if current logic is correct/complete, because it doesn't check case when other mod added prefix with higher priority which returns bool (returning false will skip other prefixes and original method call)

**Comment by pcfantasy** [comment]
OK, I see, I do not have that case. It is just a normal postfix.

I will do more test on this to see whether it will happen again.

**Comment by krzychu124** [comment]
I think that everything should work for postfix and prefix even if you'd see conflict dialog.

**Comment by pcfantasy** [comment]
[TMPE.log](https://github.com/CitiesSkylinesMods/TMPE/files/4259406/TMPE.log)
[output_log(6).txt](https://github.com/CitiesSkylinesMods/TMPE/files/4259407/output_log.6.txt)


@krzychu124 @aubergine10 

I update CSUR toolbox, some users sometime will report mod conflict.

Both CSUR toolbox and TMPE use harmony to patch  RoadBaseAI.UpdateLanes now. This confict should not happen, I do not know why.

https://github.com/citiesskylines-csur/CSURToolBox/blob/master/Patch/RoadBaseAIUpdateLanesPatch.cs

**Comment by pcfantasy** [comment]
open a new issue   #747
